### ISSUE TITLE
Scope: Uniform Buffers not bound when entering scope?

### ISSUE THREAD

**Issue by einarf** [moderngl/moderngl#336]
**In current 5.5.2**

According to the scope docs, buffers should be automatically bound with `bind_to_uniform_block` when entering the scope.
https://moderngl.readthedocs.io/en/stable/reference/scope.html

The example below will not render anything before you uncomment the two `bind_to_uniform_block` lines. As far as I understood the scope should deal with this, but that doesn't seem to be the case.

```python
from pyrr import Matrix44, matrix44, Vector3

import moderngl
import moderngl_window as mglw
from moderngl_window import geometry


class ScopeTest(mglw.WindowConfig):

    def __init__(self, **kwargs):
        super().__init__(**kwargs)

        self.prog = self.ctx.program(
            vertex_shader='''
                #version 330

                in vec3 in_position;

                uniform Projection {
                    uniform mat4 matrix;
                } proj;

                uniform View {
                    uniform mat4 matrix;
                } view;

                void main() {
                    gl_Position = proj.matrix * view.matrix * vec4(in_position, 1.0);
                }
            ''',
            fragment_shader='''
                #version 330

                out vec4 color;

                void main() {
                    color = vec4(1.0);
                }
            '''
        )
        self.vao = geometry.cube(size=(2, 2, 2)).instance(self.prog)

        self.m_proj = matrix44.create_perspective_projection_matrix(
            75, self.wnd.aspect_ratio,  # fov, aspect
            0.1, 100.0,  # near, far
            dtype='f4',
        )

        proj_uniform = self.prog['Projection']
        view_uniform = self.prog['View']
        proj_uniform.binding = 1
        view_uniform.binding = 2

        self.proj_buffer = self.ctx.buffer(reserve=proj_uniform.size)
        self.view_buffer = self.ctx.buffer(reserve=view_uniform.size)

        # self.proj_buffer.bind_to_uniform_block(1)
        # self.view_buffer.bind_to_uniform_block(2)

        self.proj_buffer.write(self.m_proj.tobytes())

        self.scope = self.ctx.scope(
            self.ctx.screen,
            enable_only=moderngl.CULL_FACE | moderngl.DEPTH_TEST,
            uniform_buffers=[
                (self.proj_buffer, 1),
                (self.view_buffer, 2),
            ],
        )

    def render(self, time=0.0, frametime=0.0, target: moderngl.Framebuffer = None):
        self.ctx.enable_only(moderngl.CULL_FACE | moderngl.DEPTH_TEST)

        m_rot = Matrix44.from_eulers(Vector3((time, time, time)))
        m_trans = matrix44.create_from_translation(Vector3((0.0, 0.0, -3.0)))
        m_modelview = matrix44.multiply(m_rot, m_trans)

        self.view_buffer.write(m_modelview.astype('f4').tobytes())

        with self.scope:
            self.vao.render(mode=moderngl.TRIANGLES)


if __name__ == '__main__':
    mglw.run_window_config(ScopeTest)
```

**Comment by einarf** [comment]
Here was the UBO example I discovered the issue. I was trying to create a scene rendered for moderngl_window sharing one UBO for projection and view matrix to optimize things (avoiding having to `Uniform.write` for every program.

**Comment by cprogrammer1994** [comment]
Thank you for the issue, the example made it trivial to spot the problem.
the buffer object and binding index were swapped.
fix is available on the master

**Comment by einarf** [comment]
Awesome. This fix alone will make a huge difference in 5.5.x
